clang/lib/CConv/ConstraintResolver.cpp

```diff
@@ -449,19 +432,6 @@ std::set<ConstraintVariable *>
         std::set<ConstraintVariable *> CVars =
             getAllSubExprConstraintVars(SubExprs);
         return addAtomAll(CVars, CS.getArr(), CS);
-      } else if (ILE->getType()->isStructureType()) {
-        // Struct initialization is treated as a series of assignments to the
-        // fields of the struct.
-        const RecordDecl *Definition =
-            ILE->getType()->getAsStructureType()->getDecl()->getDefinition();
-        int initIdx = 0;
-        for (const auto &D : Definition->fields()) {
-          std::set<ConstraintVariable *> DefCVars = Info.getVariable(D, Context);
-          Expr *InitExpr = ILE->getInit(initIdx);
-          std::set<ConstraintVariable *> InitCVars = getExprConstraintVars(InitExpr);
-          constrainLocalAssign(nullptr, D, InitExpr);
-          initIdx++;
-        }
       }
```

Comment:
Perhaps there should be an assertion the type is an array, and not something else, at this point?

Reply:
When compound literals are incorporated, that assertion would be wrong. At this point though, it could be added.

Reply:
Sounds good!